examples/scripts/coupled_variable_example.py

```diff
@@ -0,0 +1,103 @@
+"""
+Example script demonstrating how to add new variables to a model
+using CoupledVariable from an external dictionary.
+
+The dictionary format uses PyBaMM's serialization format for the expression,
+with a CoupledVariable placeholder that references the dependent variable.
+"""
+
+import pybamm
+from pybamm.expression_tree.operations.serialise import (
+    convert_symbol_from_json,
+    convert_symbol_to_json,
+)
+
+
+def add_coupled_variables_from_dict(model, coupled_vars_dict):
+    """
+    Add coupled variables to a model from an external dictionary.
+
+    Parameters
+    ----------
+    model : pybamm.BaseModel
+        The model to add coupled variables to
+    coupled_vars_dict : dict
+        Dictionary mapping new variable names to serialized expressions.
+        Expressions should use CoupledVariable nodes to reference model variables.
+    """
+    for new_var_name, serialized_expr in coupled_vars_dict.items():
+        # Deserialize the expression (CoupledVariable nodes are preserved)
+        new_var_expr = convert_symbol_from_json(serialized_expr)
+
+        # Replace CoupledVariable nodes with the actual model variables
+        def replace_coupled_vars(expr):
+            if isinstance(expr, pybamm.CoupledVariable):
+                depends_on_name = expr.name
+                if depends_on_name in model.variables:
+                    return model.variables[depends_on_name]
+                raise ValueError(f"Variable '{depends_on_name}' not found in model")
+            elif hasattr(expr, "children") and expr.children:
+                new_children = [replace_coupled_vars(c) for c in expr.children]
+                return expr.create_copy(new_children=new_children)
+            return expr
+
+        new_var_expr = replace_coupled_vars(new_var_expr)
+
+        # Add the new variable to the model
+        model.variables[new_var_name] = new_var_expr
```

Comment:
I think this function should happen somewhere in pybamm, it shouldn't be the users responsibility to define this.